docker/Dockerfile

```diff
@@ -71,6 +81,43 @@ RUN mkdir /opt/aarch64-toolchain && curl $AARCH64_TOOLCHAIN_LINK | tar xJ -C /op
 # Add generic non-root user
 RUN addgroup bao && adduser -disabled-password --ingroup bao bao
 
+# Add qemu-riscv64-virt
+RUN mkdir /opt/qemu-riscv64-virt && \
+    git clone https://github.com/qemu/qemu.git /opt/qemu-riscv64-virt --depth 1 --branch v7.2.0 && \
+    cd /opt/qemu-riscv64-virt/ && ./configure --target-list=riscv64-softmmu --enable-slirp && \
+    make -j$(nproc) && \
+    make install
+
+# Add OpenSBI
+RUN mkdir /opt/OpenSBI && git clone https://github.com/bao-project/opensbi.git /opt/OpenSBI --depth 1 --branch bao/demo
+
+# Add qemu-aarch64-virt
+RUN mkdir /opt/qemu-aarch64-virt && \
+    git clone https://github.com/qemu/qemu.git /opt/qemu-aarch64-virt --depth 1 --branch v7.2.0 && \
+    cd /opt/qemu-aarch64-virt && ./configure --target-list=aarch64-softmmu --enable-slirp && \
+    make -j$(nproc) && \
+    make install
+
+# Add U-Boot
+RUN mkdir /opt/u-boot && \
+    git clone https://github.com/u-boot/u-boot.git /opt/u-boot --depth 1 --branch v2022.10 && \
+    cd /opt/u-boot && \
+    make qemu_arm64_defconfig && \
+    echo "CONFIG_TFABOOT=y" >> .config && echo "CONFIG_SYS_TEXT_BASE=0x60000000" >> .config && \
+    make -j$(nproc) CROSS_COMPILE=/opt/aarch64-toolchain/bin/aarch64-none-elf-
+
+# Add TrustedFirmware-A
+RUN mkdir /opt/arm-trusted-firmware && \
+    git clone https://github.com/bao-project/arm-trusted-firmware.git /opt/arm-trusted-firmware\
+        --branch bao/demo --depth 1 && \
+    cd /opt/arm-trusted-firmware && \
+    make PLAT=qemu bl1 fip BL33=/opt/u-boot/u-boot.bin QEMU_USE_GIC_DRIVER=QEMU_GICV3\
+     CROSS_COMPILE=/opt/aarch64-toolchain/bin/aarch64-none-elf- && \
+    dd if=/opt/arm-trusted-firmware/build/qemu/release/bl1.bin\
+        of=/opt/arm-trusted-firmware/flash.bin && \
+    dd if=/opt/arm-trusted-firmware/build/qemu/release/fip.bin\
+        of=/opt/arm-trusted-firmware/flash.bin seek=64 bs=4096 conv=notrunc
```

Comment:
I don't see why we need to build u-boot and arm trusted firmware :thinking:

Reply:
Agree, removed u-boot and ATF build. Thanks!